### PR TITLE
Add benchmarks for pallet-membership

### PR DESCRIPTION
**Pull request by kianenigma**
- [x] TODOS: 
```
	// TODO: this can use bounded-vec as well.
	// TODO: MembershipChanged could become weight-aware
```

I think both should be follow ups though. 

- [x] polkadot companion: https://github.com/paritytech/polkadot/pull/2895
- [x] can't get the build to work :\

### CODE REVIEW

File: frame/membership/src/lib.rs [hunk 1]
```diff
@@ -282,6 +308,138 @@ impl<T: Config<I>, I: Instance> SortedMembers<T::AccountId> for Module<T, I> {
...
+	use frame_system::RawOrigin;
+	use frame_support::{traits::EnsureOrigin, assert_ok};
+	use frame_benchmarking::{benchmarks, whitelist, account, impl_benchmark_test_suite};
+
+	const SEED: u32 = 0;
+
+	// TODO: this can use bounded-vec as well.
+	// TODO: MembershipChanged could become weight-aware.
+
+	benchmarks! {
```
Comment (thiolliere) [inline frame/membership/src/lib.rs:323]:
```suggestion
	benchmarks_instance! {
```

Comment (kianenigma) [reply frame/membership/src/lib.rs:323]: ahh so that is probably why my build is all broken 🤦

**Comment by kianenigma** [comment]
\benchmark runtime pallet pallet_membership

File: frame/membership/src/lib.rs [hunk 2]
```diff
@@ -282,6 +314,152 @@ impl<T: Config<I>, I: Instance> SortedMembers<T::AccountId> for Module<T, I> {
...
+	use super::{*, Module as Membership};
+	use frame_system::RawOrigin;
+	use frame_support::{traits::EnsureOrigin, assert_ok};
+	use frame_benchmarking::{benchmarks_instance, whitelist, account, impl_benchmark_test_suite};
+
+	const SEED: u32 = 0;
+
+	// TODO: this can use bounded-vec as well.
+	// TODO: MembershipChanged could become weight-aware.
+
```
Comment (kianenigma) [inline frame/membership/src/lib.rs:328]:
```suggestion
```

File: frame/membership/src/lib.rs [hunk 3]
```diff
@@ -264,6 +285,17 @@ impl<T: Config<I>, I: Instance> Module<T, I> {
 			}
 		}
 	}
+
+	fn maybe_warn_max_members(members: &[T::AccountId]) {
+		if members.len() as u32 > T::MaxMembers::get() {
+			log::error!(
+				target: "runtime::membership",
+				"max weight exceeded [{} > {}].",
```
Comment (thiolliere) [inline frame/membership/src/lib.rs:293]: maybe a more explanatory error message,
```suggestion
				"maximum number of members used for weight is exceeded, weights can be underestimated [{} > {}].",
```

**Review by thiolliere — APPROVED** [review]
looks good to me

**Review by shawntabrizi — APPROVED** [review]

**Review by shawntabrizi — APPROVED** [review]

**Comment by shawntabrizi** [comment]
@kianenigma should be good to go here :)

**Comment by kianenigma** [comment]
bot merge

**Pull request merged**